broadcast_test.go

```diff
@@ -56,9 +56,14 @@ func testMessageMarshal(t *testing.T, m proto.Message) {
 
 // Ensure that BroadcastReceiver can register a BroadcastHandler.
 func TestBroadcast_BroadcastReceiver(t *testing.T) {
+	path, err := ioutil.TempDir("", "pilosa-")
```

Comment:
setting up new servers/clusters/etc. for testing should be handled by the `pilosa/test` package. It may need to be modified slightly to support this case, but let's not have pretty generic setup code like this in tests.

Reply:
It is possible that the holder dir and cluster dir is different; e.g., in `test/holder.go` holder path is set directly. So I think the implementation in this PR is the safest one. Is there anything wrong with expanding the holder and cluster paths in `NewServer` ?

Reply:
There's only one path - the holder path and cluster path should never be different - the fact that there are two separate objects for holder and cluster that each have a path is an unfortunate implementation detail. We should only be calling expandDir once on the path that is passed to Server via OptServerDataDir, and setting it on the holder and cluster should be fully opaque to the user. tests may be a special case, but I'm not too concerned about that.

I don't care if we do the setup in the Opt or in NewServer.